src/ui/views/EditAwards.football.tsx

```diff
@@ -0,0 +1,252 @@
+import SelectReact from "../components/SelectMultiple";
+const EditAwardsFootball = ({
+	godMode,
+	players,
+	awards,
+	season,
+}: View<"editAwards">) => {
+	useTitleBar({
+		title: "Edit awards",
+		dropdownView: "edit_awards",
+		dropdownFields: { seasons: season },
+	});
+
+	const awardsInitial = awards;
+	const [state, setState] = useState(() => {
+		const aws = helpers.deepCopy(awards);
+
+		return {
+			aws,
+		};
+	});
+	const handleChange = (obj: any, pl: any) => {
+		let error: boolean = false;
+		const p: any = pl;
+		const type = obj.props.award;
+		const numberTeam = obj.props.teamNumber;
+		const numberPlayer = obj.props.playerNumber;
+		if (state.aws == undefined) {
+			setState(prevState => {
+				const aws = helpers.deepCopy(awards);
+				return {
+					...prevState,
+					aws,
+				};
+			});
+		}
+		setState(prevState => {
+			const aws: any = prevState.aws;
+			if (p.currentStats == undefined) {
+				p.currentStats = { keyStats: "No stats available" };
+			}
+			if (
+				type == "finalsMvp" ||
+				type == "mvp" ||
+				type == "droy" ||
+				type == "oroy" ||
+				type == "dpoy"
+			) {
+				aws[type] = {
+					pid: p.pid,
+					name: p.name,
+					tid: p.tid,
+					abbrev: p.abbrev,
+					keyStats: p.currentStats.keyStats,
+				};
+			} else if (type == "allDefensive" || type == "allLeague") {
+				let arrayPids: number[] = [];
+				aws[type].map((team: any, index: number) => {
+					team["players"].map((element: Player, index: number) => {
+						arrayPids.push(element.pid);
+					});
+				});
+				if (arrayPids.includes(p.pid)) {
+					logEvent({
+						type: "error",
+						text: "Player already in teams of this category",
+						saveToDb: false,
+						persistent: true,
+					});
+					error = true;
+					obj.setValue(obj.state.select.value);
+					return {
+						...prevState,
+						aws,
+					};
+				}
+
+				aws[type][numberTeam]["players"][numberPlayer] = {
+					pid: p.pid,
+					name: p.name,
+					tid: p.tid,
+					abbrev: p.abbrev,
+					keyStats: p.currentStats.keyStats,
+				};
+			} else if (type == "allRookie") {
+				let arrayPids: number[] = [];
+				aws[type].map((element: any) => {
+					arrayPids.push(element.pid);
+				});
+				if (arrayPids.includes(p.pid)) {
+					logEvent({
+						type: "error",
+						text: "Player already in teams of this category",
+						saveToDb: false,
+						persistent: true,
+					});
+					error = true;
+					obj.setValue(obj.state.select.value);
+					return {
+						...prevState,
+						aws,
+					};
+				}
+				aws[type][numberPlayer] = {
+					pid: p.pid,
+					name: p.firstName + " ",
+					tid: p.tid,
+					abbrev: p.abbrev,
+					keyStats: p.currentStats.keyStats,
+				};
+			}
+
+			return {
+				...prevState,
+				aws,
+			};
+		});
+		return error;
+	};
+
+	const handleFormSubmit = async (event: FormEvent) => {
+		event.preventDefault();
+		setState(prevState => ({
+			...prevState,
+			saving: true,
```

Comment:
Ah I see, so `state` has two properties. Would make more sense to keep them separate, you can make as many state variables as you want with hooks. LIke above do:

`const [saving, setSaving] = useState(false);`

and then this can just be:

`setSaving(true);`